cyclops-ui/src/components/pages/Modules/Modules.tsx

```diff
@@ -49,6 +71,7 @@ const Modules = () => {
         .get(`/api/modules/list`)
         .then((res) => {
           setAllData(res.data);
+          populateNamespaceData(res.data);
```

Comment:
Move the call of this function outside the `fetchModules` loop.

In this useEffect, before the `fetchModules` function, create a call towards `/api/namespaces`. This will return all available namespaces with which you can populate both the `namespaceFilterData` and the `moduleNamespaceFilter`